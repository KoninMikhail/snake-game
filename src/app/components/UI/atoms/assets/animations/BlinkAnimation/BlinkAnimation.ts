import styled, { keyframes } from 'styled-components';
import { IAnimationProps } from '@ui/atoms/assets/animations/Animation.interface';

const blinkAnimationKeyframes = keyframes`
  50% {
    opacity: 0.0;
  }
`;

/**
 * @name BlinkAnimation
 * @author Mikhail Konin <dev.konin@gmail.com>
 *
 * Use for creation blink-effect.
 *
 * @param delayBeforeChange - animation delayBeforeChange
 * @return {JSX.Element}
 */
export const BlinkAnimation = styled.div<IAnimationProps>`
    animation-duration: ${({ theme }) => theme.animations.blink.duration.toString() + 'ms'};
    animation-fill-mode: ${({ theme }) => theme.animations.blink.fill};
    animation-direction: normal;
    animation-name: ${blinkAnimationKeyframes};
    animation-delay: ${({ delay = 0 }) => delay.toFixed().toString() + 'ms'};
    animation-iteration-count: ${({ theme }) => theme.animations.blink.count};
    animation-timing-function: ${({ theme }) => theme.animations.blink.timing};
`;
