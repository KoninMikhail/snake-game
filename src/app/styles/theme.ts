import {colors, ThemeColors} from './theme_parts/colors';
import {gaps, ThemeGaps} from './theme_parts/gaps';
import {icons, ThemeIcons} from './theme_parts/icons';
import {ThemeTypography, typography} from './theme_parts/typography';
import {animations, ThemeAnimations} from '@styles/theme_parts/animations';

/**
 * @typedef IBaseTheme
 */
export interface IBaseTheme {
    animations: ThemeAnimations;
    colors: ThemeColors;
    gaps: ThemeGaps;
    icons: ThemeIcons;
    typography: ThemeTypography;
}

/**
 * Theme
 */
export const theme: IBaseTheme = {
    colors,
    animations,
    gaps,
    icons,
    typography,
};
