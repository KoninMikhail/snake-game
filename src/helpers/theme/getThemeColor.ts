import { ThemeColors } from '@styles/theme_parts/colors';
import { IBaseTheme } from '@styles/theme';

/**
 * @name getThemeColor
 *
 * Get the color from theme.
 *
 * @param theme - The theme object
 * @param color - The color you want to get from the theme.
 * @returns A function that takes a theme and a color and returns the color from the theme.
 */
const getThemeColor: GetFromThemeFn<IBaseTheme, ThemeColors> = (theme, color) => {
    return theme.colors[color];
};

export default getThemeColor;
