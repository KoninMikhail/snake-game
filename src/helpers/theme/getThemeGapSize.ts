import {IBaseTheme} from '@styles/theme';
import {ThemeGaps} from '@styles/theme_parts/gaps';

/**
 * @name getThemeGapSize
 *
 * Get the gap size from the theme.
 *
 * @param theme - The theme object
 * @param size - The size of the gap you want to get.
 * @returns A function that takes a theme and a size and returns the theme.gaps[size]
 */
const getThemeGapSize: GetFromThemeFn<IBaseTheme, ThemeGaps> = (theme, size) => theme.gaps[size];

export default getThemeGapSize;
