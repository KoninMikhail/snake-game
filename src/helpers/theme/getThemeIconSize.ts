import { IBaseTheme } from '@styles/theme';
import { ThemeIcons } from '@styles/theme_parts/icons';

/**
 * @name getThemeIconSize
 *
 * Get the icon size from the theme.
 *
 * @param theme - The theme object
 * @param size - The size of the icon you want to get.
 * @returns A function that takes a theme and a size and returns the theme.icons[size]
 */
const getThemeIconSize: GetFromThemeFn<IBaseTheme, ThemeIcons> = (theme, size) => {
    return theme.icons[size];
};

export default getThemeIconSize;
