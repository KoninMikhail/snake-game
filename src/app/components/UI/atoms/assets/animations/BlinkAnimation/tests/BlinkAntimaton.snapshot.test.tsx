import { BlinkAnimation } from '../BlinkAnimation';
import { render } from '@testing-library/react';
import { IBaseTheme } from '@styles/theme';

/**
 * Mocks
 */
const fakeTheme = {
    animations: {
        blink: {
            duration: 1200,
            count: 'infinite',
            timing: 'step-start',
            fill: 'forwards',
        },
    },
} as unknown as IBaseTheme;

/**
 * Tests
 */
describe('BlinkAnimation', () => {
    it('equal snapshot', () => {
        const { container } = render(<BlinkAnimation theme={fakeTheme} />);
        expect(container).toMatchSnapshot();
    });
});
