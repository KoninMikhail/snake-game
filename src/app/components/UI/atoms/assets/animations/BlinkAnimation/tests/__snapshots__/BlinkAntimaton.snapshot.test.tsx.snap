// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlinkAnimation equal snapshot 1`] = `
.c0 {
  -webkit-animation-duration: 1200ms;
  animation-duration: 1200ms;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-name: iDIVWf;
  animation-name: iDIVWf;
  -webkit-animation-delay: 0ms;
  animation-delay: 0ms;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: step-start;
  animation-timing-function: step-start;
}

<div>
  <div
    class="c0"
  />
</div>
`;
